### PR TITLE
Add empty Memory class

### PR DESCRIPTION
**Pull request by aidanwolter3**
This class will receive the raw data form the ROM, and allow other
classes to read/write.

### CODE REVIEW

File: src/memory.h [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+#include <cstdint>
+#include <vector>
+
+class Memory {
+ public:
+  Memory(std::vector<uint8_t> data);
+  ~Memory();
+
+  uint8_t Read(uint16_t address);
+  uint8_t Write(uint16_t address);
```
Comment (Colemalban) [inline src/memory.h]: What does the |uint8_t| return value mean? and what data is being written?

File: src/BUILD [hunk 2]
```diff
@@ -5,6 +5,8 @@ cc_library(
     ],
     srcs = [
         "atlas.cc",
+        "memory.cc",
+        "memory.h",
```
Comment (Colemalban) [inline src/BUILD:9]: Should the header be made public for this lib?

File: src/memory.cc [hunk 3]
```diff
@@ -0,0 +1,15 @@
+#include "memory.h"
+
+Memory::Memory(std::vector<uint8_t> data)
+    : data_(std::move(data)) {}
+
+Memory::~Memory() = default;
+
+Memory::Status Memory::Read(uint16_t address, uint8_t* byte) {
+  *byte = data_[address];
```
Comment (Colemalban) [inline src/memory.cc:9]: Could |byte| be a null pointer or something weird like that?

File: src/memory.h [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+#include <cstdint>
+#include <vector>
+
+class Memory {
+ public:
+  Memory(std::vector<uint8_t> data);
+  ~Memory();
+
+  enum Status : uint8_t {
```
Comment (Colemalban) [inline src/memory.h:12]: Should we use an enum class?

**Review by Colemalban — APPROVED** [review]

File: src/memory.h (hunk 4 above)
Comment (aidanwolter3) [reply src/memory.h:12]: great idea!

File: src/memory.cc (hunk 3 above)
Comment (aidanwolter3) [reply src/memory.cc:9]: Yeah, I'll check for nullptr and add some tests.

**Pull request merged**